src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -76,7 +76,7 @@ public IPropertySetterAnyValueConfiguration<TValue> CallToSet<TValue>(Expression
                     this.CreateVoidArgumentValidationConfiguration(fake, newParsedSetterCallExpression));
         }
 
-        private static void GuardAgainstNonFake(object target)
+        private static void GuardAgainstNonFake(object? target)

```

Comment:
Too bad there isn't an attribute to say "this method will throw if the argument is null". This way the compiler will now that the argument can't be null after the call. R# has an attribute like that ([example](https://github.com/thomaslevesque/Linq.Extras/blob/528233fef751cdfefbec3bb10096b96c41cd4d9e/src/Linq.Extras/Internal/ArgumentExtensions.cs#L10)).